src/remote/action.go

```diff
@@ -257,6 +258,30 @@ func (c *Client) uploadInputDir(ch chan<- *uploadinfo.Entry, target *core.BuildT
 			}
 			continue
 		}
+		if i, ok := input.(core.SubrepoFileLabel); ok && !target.Local {
+			for _, p := range i.Paths(c.state.Graph) {
+				subrepoPath, err := filepath.Rel(target.Subrepo.PackageRoot, p)
+				if err != nil {
+					return nil, fmt.Errorf("%v: source file not in subrepo package root (%v): %v", target.Label, p, err)
+				}
+				fileNode, dirNode, symlinkNode, err := remotefs.FindNode(target.Subrepo.FS(), subrepoPath)
+				if err != nil {
+					return nil, fmt.Errorf("%v: failed to find file in subrepo output: %v", target.Label, err)
+				}
+
+				dir := b.Dir(filepath.Dir(p))
+				if fileNode != nil {
+					dir.Files = append(dir.Files, fileNode)
+				}
+				if dirNode != nil {
+					dir.Directories = append(dir.Directories, dirNode)
+				}
+				if symlinkNode != nil {
+					dir.Symlinks = append(dir.Symlinks, symlinkNode)
+				}
+			}
+			continue
```

Comment:
Do we need this continue here?

Reply:
Yup otherwise it will try and add this via `uploadInput()` on line 285.